Description: Patch adds keys '1', '3' in replay mode to move marks by +/-5secs.
             If the next mark is nearer than 5secs, move the mark on the same position.
Author: FrankJepsen at vdrportal.de
Origin: http://vdrportal.de/board/thread.php?postid=718583#post718583

Index: vdr/menu.c
===================================================================
--- vdr.orig/menu.c	2015-02-07 11:23:12.624741872 +0100
+++ vdr/menu.c	2015-02-07 11:23:12.624741872 +0100
@@ -5493,10 +5493,47 @@
                    Stop();
                    return osEnd;
     default: {
+      bool play, forward;
+      int speed;
+      int Current, Total;
+      cMark *m;
+
       DoShowMode = false;
       switch (int(Key)) {
         // Editing:
         case kMarkToggle:      MarkToggle(); break;
+        case k1|k_Repeat:
+        case k1:               if (GetReplayMode(play, forward, speed) && !play && GetIndex(Current, Total) && (m = marks.Get(Current)) != NULL) {
+                                  displayFrames = true;
+                                  int p = SkipFrames(-SecondsToFrames(5, FramesPerSecond()));
+                                  cMark *m2;
+                                  while ((m2 = marks.Prev(m)) != NULL && m2->Position() == m->Position())
+                                        m = m2;
+                                  if ((m2 = marks.Prev(m)) != NULL && m2->Position() > p)
+                                     p = m2->Position();
+                                  m->SetPosition(p);
+                                  Goto(m->Position(), true);
+                                  marksModified = true;
+                                  }
+                               else
+                                  SkipSeconds(-20);
+                               break;
+        case k3|k_Repeat:
+        case k3:               if (GetReplayMode(play, forward, speed) && !play && GetIndex(Current, Total) && (m = marks.Get(Current)) != NULL) {
+                                  displayFrames = true;
+                                  int p = SkipFrames(SecondsToFrames(5, FramesPerSecond()));
+                                  cMark *m2;
+                                  while ((m2 = marks.Next(m)) != NULL && m2->Position() == m->Position())
+                                        m = m2;
+                                  if ((m2 = marks.Next(m)) != NULL && m2->Position() < p)
+                                     p = m2->Position();
+                                  m->SetPosition(p);
+                                  Goto(m->Position(), true);
+                                  marksModified = true;
+                                  }
+                               else
+                                  SkipSeconds(20);
+                               break;
         case kPrev|k_Repeat:
         case kPrev:
         case kMarkJumpBack|k_Repeat:
